### DISCUSSION TITLE
关于使用Mvvm中的一些疑问

### DISCUSSION THREAD

**Discussion by xzf888** [BYJRK/DotNet-Discussions#72] [MVVM]
### 1. textbox绑定的变量，如何触发变量改变事件，（如变量为空自动给一个默认值），

是不是[ObservableProperty]这个就没用了
目前我的做法是
![image](https://user-images.githubusercontent.com/48784071/230009895-e5c74a64-f332-43ee-a047-b2810d1fe3df.png)
这样的话，**大部分控件都需要写 private和public get set，那[ObservableProperty]这个不就没用了吗**

**Comment by xzf888** [comment]
第一个问题看了下微软文档，发现有 OnNameChanging(value)和 OnNameChanged(value)
可以改成这样写
![image](https://user-images.githubusercontent.com/48784071/230019010-31800f78-e95e-42fb-b77d-cc5b869339dd.png)

**Comment by BYJRK** [comment]
你有看过我的视频吗，介绍Toolkit的
https://www.bilibili.com/video/BV12x4y177qB

**Reply by xzf888** [reply]
就是看的这个视频慢慢学的，但是没找到怎么解决文件拖放的办法

**Reply by BYJRK** [reply]
Drop事件如果你不需要和后台逻辑打交道，可以放在View中实现；否则的话，你可以用Interaction.Trigger来绑定到RelayCommand，或者自己写一个附加属性或行为来实现。

**Reply by xzf888** [reply]
已经这样写了，但是还是不支持拖放
![image](https://user-images.githubusercontent.com/48784071/230088110-f2e72723-adc3-4396-a2e0-78142f5d18d9.png)
![image](https://user-images.githubusercontent.com/48784071/230088480-670b829f-82ec-4c16-9eb0-ea9c0d427ec0.png)

![image](https://user-images.githubusercontent.com/48784071/230088282-b89a2966-e931-4c2c-ba18-874e5f741ebd.png)

**Reply by BYJRK** [reply]
是不是没有开AllowDrop之类？

**Reply by xzf888** [reply]
我给textbox开了AllowDrop,你意思是给整个窗口开AllowDrop吗?

**Comment by PQGit** [comment]
把Drag追加下应该就好啦。
